snappiershot/snapshot/metadata.py

```diff
@@ -43,7 +97,26 @@ def matches(self, metadata_dict: Dict) -> bool:
         Args:
             metadata_dict: The "metadata" section to compare against this object.
         """
-        return self.caller_info.args == metadata_dict["arguments"]
+        # Loop through every value in the metadata dictionary
+        metadata_args = self.caller_info.args
+        for key in metadata_args.keys():
+            # Initialize objects
+            metadata_obj = metadata_args.get(key)
+            dict_from_file = metadata_dict["arguments"].get(key)
+
+            # If the object is a list, must loop through it and compare each index
+            if isinstance(metadata_obj, list) and isinstance(dict_from_file, list):
```

Comment:
Could either be a `Sequence` instead of a list?

Reply:
Also, if `dict_from_file` could be a `list`, using "dict" in the name seems misleading

Reply:
[removed need to loop through list](https://github.com/MORSECorp/snappiershot/pull/96/commits/19e03c7a0800ba04bc8faf6b6dd656f4d29da1b8)